### ISSUE TITLE
Remove runtime support for hierarchical references

### ISSUE THREAD

**Issue by lhstrh** [icyphy/reactor-ts#53]
We currently have runtime support for handling arguments of the form `<reactor>.<port>`. This complicates things a bit, and muddles the implementation. It's solely to accommodate the use of notation that is common in LF. It occurred to me that we don't actually need this. For example, the following code:
```
state violation_detected:boolean(false);
reaction(startup) -> d.x {=
...
```
currently gets transformed into:
```
this.addReaction(
            new Triggers(this.startup),
            new Args(this.violation_detected, { x: this.getWriter(this.d.x) }),
            function (this, __violation_detected: State<boolean>, __d: { x: ReadWrite<number> }) {
                // =============== START react prologue
                const util = this.util;
                let violation_detected = __violation_detected.get();
                let d = {x: __d.x.get()};
...
```
But this could just as well be translated as follows:
```
this.addReaction(
            new Triggers(this.startup),
            new Args(this.violation_detected, this.getWriter(this.d.x)),
            function (this, __violation_detected: State<boolean>, __x: ReadWrite<number>) {
                // =============== START react prologue
                const util = this.util;
                let violation_detected = __violation_detected.get();
                let d = {x: __x.get()};
...
```
TLDR: the wrapping to obtain a reference of the form `<reactor>.<port>` can be done in the prologue; it doesn't also have to happen in the argument list.

**Comment by mew2ub** [comment]
Yeah, this change could work. We probably did it that way because there was no prologue when we were working on hierarchical references.

**Comment by lhstrh** [comment]
This is done.
